services/motion/builtin/builtin.go

```diff
@@ -282,14 +282,24 @@ func (ms *builtIn) MoveOnGlobe(
 		return false, errors.New("destination cannot be nil")
 	}
 
-	moveRequest, err := ms.newMoveOnGlobeRequest(ctx, componentName, destination, movementSensorName, obstacles, motionCfg, extra)
+	mr, err := ms.newMoveOnGlobeRequest(ctx, componentName, destination, movementSensorName, obstacles, motionCfg, extra)
 	if err != nil {
 		return false, err
 	}
 
+	maxReplans := -1
+	i := 0
```

Comment:
can we please give this a more descriptive name than `i`?

If this represents a replanCould could we please call it `replanCount`?

Could we please also move these variables on to some struct

Reply:
Happy to rename.

Why would we want them on a struct when they're relevant only in this scope for this one loop?

Reply:
fair enough, nix the struct but lets give it a better name.